feature_tests/kotlin/somelib/src/main/kotlin/dev/diplomattest/somelib/NestedBorrowedFields.kt

```diff
@@ -0,0 +1,41 @@
+package dev.diplomattest.somelib
+
+import com.sun.jna.Library
+import com.sun.jna.Native
+import com.sun.jna.Pointer
+import com.sun.jna.Structure
+
+internal interface NestedBorrowedFieldsLib: Library {
+}
+
+class NestedBorrowedFieldsNative: Structure(), Structure.ByValue {
+    @JvmField
+    var fields: BorrowedFieldsNative = BorrowedFieldsNative();
+    @JvmField
+    var bounds: BorrowedFieldsWithBoundsNative = BorrowedFieldsWithBoundsNative();
+    @JvmField
+    var bounds2: BorrowedFieldsWithBoundsNative = BorrowedFieldsWithBoundsNative();
+  
+    // Define the fields of the struct
+    override fun getFieldOrder(): List<String> {
+        return listOf("fields", "bounds", "bounds2")
+    }
+}
+
+class NestedBorrowedFields internal constructor (
+    internal val nativeStruct: NestedBorrowedFieldsNative,
+    internal val xEdges: List<Any>,
+    internal val yEdges: List<Any>,
+    internal val zEdges: List<Any>
+    ) {
+    val fields: BorrowedFields = BorrowedFields(nativeStruct.fields, listOf())
```

Comment:
issue: the borrow edges aren't getting passed down, I think? Is borrowing a WIP?

(that's fine if it is, just want to know)

Reply:
hmm I thought I had that. But I was just looking at the `Foo.extractFromBounds` (But of course would appreciate a check on the lifetimes in method params)... I think I understand now. I only handled borrows for method params but not for struct fields. This was just an oversight on my part.

Reply:
I'll work on the struct field borrows still